### PR TITLE
Rewrite in ES2015, use Catberry 8

### PR DESCRIPTION
**Pull request by JuliaRechkunova**

### CODE REVIEW

File: catberry_components/ga/index.js [hunk 1]
```diff
@@ -1,123 +1,88 @@
...
+			}
+			this._window.ga('send', 'pageview', getLocation(event.context));
+		});
+	}
+
+	/**
+	 * Tracks errors.
+	 */
+	trackErrors() {
+		this.$context.on('error', (error) => {
```
Comment (pragmadash) [inline catberry_components/ga/index.js]: We can remove braces here.

File: catberry_components/head/index.js [hunk 2]
```diff
@@ -43,35 +8,32 @@ util.inherits(Head, ComponentBase);
...
+	render() {
+		const location = this.$context.location.clone();
+		const socialLogo = this.$context.location.clone();
+
+		location.scheme = 'http';
+		socialLogo.scheme = location.scheme;
+		socialLogo.path = '/public/assets/head/images/social-logo.png';
+
+		return this.$context.getStoreData()
+			.then((pages) => {
```
Comment (pragmadash) [inline catberry_components/head/index.js]: Can remove braces and `return` here.

File: catberry_components/pages/index.js [hunk 3]
```diff
@@ -43,25 +8,20 @@ util.inherits(Pages, ComponentBase);
...
+	/**
+	 * Gets data context for template engine.
+	 * This method is optional.
+	 * @returns {Promise<Object>|Object|null|undefined} Data context
+	 * for template engine.
+	 */
+	render() {
+		return this.$context.getStoreData()
+			.then((pages) => {
+				return this.localizeContext(pages);
```
Comment (pragmadash) [inline catberry_components/pages/index.js]: We can remove braces and `return` here.

File: catberry_components/section-documentation/index.js [hunk 4]
```diff
@@ -1,46 +1,39 @@
...
+	/**
+	 * Gets data context for template engine.
+	 * This method is optional.
+	 * @returns {Promise<Object>|Object|null|undefined} Data context
+	 * for template engine.
+	 */
+	render() {
+		return this.$context.getStoreData()
+			.then((data) => {
+				return this.localizeContext(data);
```
Comment (pragmadash) [inline catberry_components/section-documentation/index.js]: Remove braces and `return`.

File: catberry_components/section-overview/index.js [hunk 5]
```diff
@@ -1,35 +1,39 @@
...
+	/**
+	 * Gets data context for template engine.
+	 * This method is optional.
+	 * @returns {Promise<Object>|Object|null|undefined} Data context
+	 * for template engine.
+	 */
+	render() {
+		return this.$context.getStoreData()
+			.then((data) => {
+				return this.localizeContext(data);
```
Comment (pragmadash) [inline catberry_components/section-overview/index.js]: Remove braces and `return`.

File: catberry_components/section-quotes/index.js [hunk 6]
```diff
@@ -43,36 +8,31 @@ util.inherits(SectionQuotes, ComponentBase);
...
+	/**
+	 * Gets data context for template engine.
+	 * This method is optional.
+	 * @returns {Promise<Object>|Object|null|undefined} Data context
+	 * for template engine.
+	 */
+	render() {
+		return this.$context.getStoreData()
+			.then((data) => {
+				return this.localizeContext(data);
```
Comment (pragmadash) [inline catberry_components/section-quotes/index.js]: Remove braces and `return`;

File: catberry_components/section-quotes/index.js [hunk 7]
```diff
@@ -43,36 +8,31 @@ util.inherits(SectionQuotes, ComponentBase);
...
+	}
+
+	/**
+	 * Binds events.
+	 */
+	bind() {
+		const window = this.$context.locator.resolve('window');
+		window.twttr.ready()
+			.then(() => {
+				window.twttr.widgets.load();
```
Comment (pragmadash) [inline catberry_components/section-quotes/index.js]: Remove braces.

File: lib/ComponentBase.js [hunk 8]
```diff
@@ -1,73 +1,40 @@
...
-/**
- * Gets data context for template engine.
- * This method is optional.
- * @returns {Promise<Object>|Object|null|undefined} Data context
- * for template engine.
- */
-ComponentBase.prototype.render = function () {
-	return this.localizeContext();
-};
+	/**
```
Comment (pragmadash) [inline lib/ComponentBase.js]: Weird indentation, spaces?

File: lib/GitHubClient.js [hunk 9]
```diff
@@ -1,224 +1,177 @@
...
-}
+class GitHubClient {
 
-/**
- * Current logger.
- * @type {Logger}
- * @private
- */
-GitHubClient.prototype._logger = null;
+	/**
```
Comment (pragmadash) [inline lib/GitHubClient.js]: Also, looks like spaces.

File: lib/GitHubClient.js [hunk 10]
```diff
@@ -1,224 +1,177 @@
...
-			parts = urlPath.split('/'),
-			document = parts[parts.length - 1];
-
-		if (!DOCUMENTS.hasOwnProperty(document)) {
-			next();
-			return;
-		}
+			const methodName = DOCUMENTS[document];
+			this[methodName]()
+				.then((content) => {
```
Comment (pragmadash) [inline lib/GitHubClient.js]: Remove braces.

File: lib/GitHubClient.js [hunk 11]
```diff
@@ -1,224 +1,177 @@
...
+					response.send(content);
+				})
+				.catch((error) => {
```
Comment (pragmadash) [inline lib/GitHubClient.js]: Remove braces.

File: lib/GitHubClient.js [hunk 12]
```diff
@@ -1,224 +1,177 @@
...
+	getReadme() {
+		return this.request({
+			url: this._buildUrl('/repos/catberry/catberry/readme'),
+			method: 'GET',
+			headers: {
+				Accept: 'application/vnd.github.VERSION.html'
+			}
+		})
+			.then((content) => {
+				return postProcessContent(content);
```
Comment (pragmadash) [inline lib/GitHubClient.js]: Remove braces and `return`.

File: lib/GitHubClient.js [hunk 13]
```diff
@@ -1,224 +1,177 @@
...
+			url: this._buildUrl(
+					'/repos/catberry/catberry/contents/docs/index.md'
+				),
+			headers: {
+				Accept: 'application/vnd.github.VERSION.html'
+			},
+			method: 'GET'
+		})
+			.then((content) => {
+				return postProcessContent(content);
```
Comment (pragmadash) [inline lib/GitHubClient.js]: Remove braces and `return`.

File: package.json [hunk 14]
```diff
@@ -14,14 +15,14 @@
 		"type": "git",
 		"url": "git://github.com/catberry/catberry-homepage.git"
 	},
-	"author": {
-		"name": "Julia Rechkunova",
-		"email": "julia.rechkunova@gmail.com"
-	},
```
Comment (pragmadash) [inline package.json]: AFAIK, author is required field in NPM.

File: lib/ComponentBase.js (hunk 8 above)
Comment (JuliaRechkunova) [inline lib/ComponentBase.js]: Spaces were before

**Pull request merged**